### PR TITLE
Archetype changes for product branch

### PR DESCRIPTION
**Pull request by cunningt**
Combination of cherry pick of https://github.com/jboss-fuse/camel-spring-boot/commit/7376d4f2f7274603e67ba2978a9dd4f6bcdb4cba and a number of other archetype-related changes in 4.4 -> move to 4.8 branch

### CODE REVIEW

File: archetypes/camel-archetype-spring-boot/src/main/resources/archetype-resources/pom.xml [hunk 1]
```diff
@@ -30,8 +30,12 @@
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+    <maven.compiler.source>17</maven.compiler.source>
+    <maven.compiler.target>17</maven.compiler.target>
+    <maven.compiler.release>17</maven.compiler.release>
     <spring.boot-version>${spring-boot-version}</spring.boot-version>
     <surefire.plugin.version>3.0.0-M4</surefire.plugin.version>
+    <jkube.generator.from>registry.access.redhat.com/ubi8/openjdk-17:latest</jkube.generator.from>
```
Comment (mcarlett) [inline archetypes/camel-archetype-spring-boot/src/main/resources/archetype-resources/pom.xml]: do we want still use ubi8 by default or do we want to use ubi9 as default?

Comment (cunningt) [reply archetypes/camel-archetype-spring-boot/src/main/resources/archetype-resources/pom.xml]: Moving to 9 sounds fine to me - any opinions @Croway ?

Comment (mcarlett) [reply archetypes/camel-archetype-spring-boot/src/main/resources/archetype-resources/pom.xml]: moreover, I want to add that starting from OCP 4.14 the DeploymentConfig [has been deprecated](https://access.redhat.com/articles/7041372) while JKube still use DeploymentConfig as default deployment type for OCP, I would like to suggest to force the usage of Deployment instead, using property
`<jkube.enricher.jkube-controller.type>Deployment</jkube.enricher.jkube-controller.type>`

File: archetypes/camel-archetype-spring-boot/src/main/resources-filtered/META-INF/maven/archetype-metadata.xml [hunk 2]
```diff
@@ -24,12 +24,18 @@
     <requiredProperty key="camel-version">
       <defaultValue>${project.version}</defaultValue>
     </requiredProperty>
```
Comment (mcarlett) [inline archetypes/camel-archetype-spring-boot/src/main/resources-filtered/META-INF/maven/archetype-metadata.xml]: I think `camel-version`  is not needed, and the BOM's in the dependencyManagement  should be replace by 

```
<dependencyManagement>
    <dependencies>
      <dependency>
        <groupId>com.redhat.camel.springboot.platform</groupId>
        <artifactId>camel-spring-boot-bom</artifactId>
        <version>${camel-spring-boot-version}</version>
        <type>pom</type>
        <scope>import</scope>
      </dependency>
    </dependencies>
  </dependencyManagement>
```

that uses only `camel-spring-boot-version`

File: archetypes/camel-archetype-spring-boot/src/main/resources/archetype-resources/pom.xml (hunk 1 above)
Comment (Croway) [reply archetypes/camel-archetype-spring-boot/src/main/resources/archetype-resources/pom.xml]: let's move to 9, sounds good to me

**Review by Croway — APPROVED** [review]

File: archetypes/camel-archetype-spring-boot/src/main/resources/archetype-resources/pom.xml (hunk 1 above)
Comment (cunningt) [reply archetypes/camel-archetype-spring-boot/src/main/resources/archetype-resources/pom.xml]: Changed to ubi9

File: archetypes/camel-archetype-spring-boot/src/main/resources-filtered/META-INF/maven/archetype-metadata.xml (hunk 2 above)
Comment (cunningt) [reply archetypes/camel-archetype-spring-boot/src/main/resources-filtered/META-INF/maven/archetype-metadata.xml]: Changed this instance of camel-version to camel-spring-boot-version - keeping the camel-version for consistency with older archetypes though

File: archetypes/camel-archetype-spring-boot/src/main/resources/archetype-resources/pom.xml (hunk 1 above)
Comment (Croway) [reply archetypes/camel-archetype-spring-boot/src/main/resources/archetype-resources/pom.xml]: @cunningt can you add `<jkube.generator.from>registry.access.redhat.com/ubi8/openjdk-17:latest</jkube.generator.from>` as well?

**Review by mcarlett — APPROVED** [review]

**Pull request merged**